bolt-modules/boltlib/lib/puppet/datatypes/target.rb

```diff
@@ -27,7 +27,9 @@
       password => Callable[[], Optional[String[1]]],
       port => Callable[[], Optional[Integer]],
       protocol => Callable[[], Optional[String[1]]],
-      user => Callable[[], Optional[String[1]]],
+      transport => Callable[[], Optional[String[1]]],
+      transport_config => Callable[[], Optional[Hash[String[1], Data]]],
```

Comment:
I don't think the results of `transport` or `transport_config` function should be `Optional`. A target always has to have a transport, right? And `transport_config` will always be a hash, just empty if there are no options.

As opposed to the other functions, where particularly `remote` targets might _not_ have any of host/port/user/password.

Reply:
Ah, that makes sense. I thought it was just a Puppet-ism. Will update. :)